### PR TITLE
Add builder for Cairo

### PR DESCRIPTION
**Pull request by giordano**
I'm copying this from https://github.com/JuliaGraphics/CairoBuilder.  Note: building for all platforms works on Travis: https://travis-ci.org/JuliaGraphics/CairoBuilder/builds/576867465.

A release would be welcome :slightly_smiling_face:

### CODE REVIEW

File: C/Cairo/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# TODO: find out why we need to manually add "${prefix}/include/freetype2",
+# the pkg-config file for freetype2 should already have this information.
+CFLAGS="-I${prefix}/include -I${prefix}/include/freetype2"
+
+if [[ "${target}" == *-apple-* ]]; then
+    BACKEND_OPTIONS="--disable-xlib --enable-quartz"
+elif [[ "${target}" == *-mingw* ]]; then
+    BACKEND_OPTIONS="--disable-xlib --enable-win32"
+else
+    BACKEND_OPTIONS="--disable-xlib"
```
Comment (giordano) [inline C/Cairo/build_tarballs.jl]: We "only" need to build libX11 to enable the native backend on Linux (and hopefully FreeBSD, too)

File: C/Cairo/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/cairo-*/
+
+LDFLAGS="${LDFLAGS} -L${prefix}/lib"
+# TODO: find out why we need to manually add "${prefix}/include/freetype2",
+# the pkg-config file for freetype2 should already have this information.
+CFLAGS="-I${prefix}/include -I${prefix}/include/freetype2"
```
Comment (giordano) [inline C/Cairo/build_tarballs.jl]: Don't ask me why the build system can't get that it needs to use `-I${prefix}/include/freetype2`, it's in freetype2's pkg-config file :man_shrugging:  I went with the easy solution of just manually doing it.

**Comment by giordano** [comment]
PR #42 adds a builder for libX11, which hopefully will allow us to build Cairo with a native backend also on Linux/FreeBSD.  We don't strictly need it on macOS and Windows, is it possible to specify target-specific dependencies?

File: C/Cairo/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    "https://github.com/bicycle1885/ZlibBuilder/releases/download/v1.0.4/build_Zlib.v1.2.11.jl",
+    "https://github.com/JuliaIO/LibpngBuilder/releases/download/v1.0.3/build_libpng.v1.6.37.jl",
+    "https://github.com/JuliaPackaging/Yggdrasil/releases/download/Pixman-v0.36.0-0/build_Pixman.v0.36.0.jl",
+    "https://github.com/JuliaGraphics/FreeTypeBuilder/releases/download/v2.9.1-4/build_FreeType2.v2.10.0.jl",
+    "https://github.com/JuliaPackaging/Yggdrasil/releases/download/Bzip2-v1.0.6-2/build_Bzip2.v1.0.6.jl",
+    "https://github.com/giordano/Yggdrasil/releases/download/X11-v1.6.8/build_X11.v1.6.8.jl"
```
Comment (giordano) [inline C/Cairo/build_tarballs.jl]: Waiting for #42 to be merged... you already know what I'm going to say

File: C/Cairo/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply C/Cairo/build_tarballs.jl]: Done in #42!

**Comment by ararslan** [comment]
This is truly a momentous and long-awaited occasion

**Pull request merged**